lib/npg_tracking/data/transcriptome/find.pm

```diff
@@ -2,13 +2,43 @@ package npg_tracking::data::transcriptome::find;
 
 use Moose::Role;
 use Carp;
-
 use npg_tracking::util::abs_path qw(abs_path);
 
+Readonly::Scalar our $DEFAULT_ALIGNER => q[tophat2];
+Readonly::Scalar our $DEFAULT_QUANTIFIER => q[salmon];
+Readonly::Scalar our $DEFAULT_ANALYSIS => $DEFAULT_ALIGNER;
+
 with qw/ npg_tracking::data::reference::find /;
 
 our $VERSION = '0';
 
+has 'analysis' => (default  => $DEFAULT_ANALYSIS,
+                   is       => 'ro',
+                   required => 0,);
+
+has '_file_type' => (isa      => q{Maybe[Str]},
+                     is       => q{ro},
+                     required => 0,
+                     writer   => q{_set_file_type},);
+
+has '_subfolder' => (isa      => q{Maybe[Str]},
+                     is       => q{ro},
+                     required => 0,
+                     writer   => q{_set_subfolder},);
+
+has '_index_name_ext' => (isa      => q{Str},
+                          is       => q{ro},
+                          required => 0,
+                          lazy     => 1,
+                          builder  => q{_build_index_name_ext},);
+
+sub _build_index_name_ext {
+    my $self = shift;
+    my $ext = $self->analysis eq $DEFAULT_ALIGNER ? q[bt2] :
+              $self->analysis eq $DEFAULT_QUANTIFIER ? q[json] : return;
```

Comment:
Would be cleaner as 
my $ext = $self->analysis eq $DEFAULT_ALIGNER ? q[bt2] :
      ($self->analysis eq $DEFAULT_QUANTIFIER ? q[json] : undef);